Utilities/Remove-OldAzureVMs.ps1

```diff
@@ -164,16 +165,48 @@ if ($RemoveAutoCleanupVMs) {
 $cleanupRGScriptBlock = {
     $RGName = $args[0]
     $DryRun = $args[1]
+    $RemoveDisks = $args[2]
     if (-not $DryRun) {
+        if ($RemoveDisks) {
+            $vms = Get-AzVM -ResourceGroupName $RGName
+        }
         Remove-AzResourceGroup -Name $RGName -Verbose -Force
+
+        if ($? -and $RemoveDisks) {
+            foreach ($vm in $vms) {
+                $osDiskUri = $vm.StorageProfile.OSDisk.Vhd.Uri
+                #VHD URL example - http://storageaccountname.blob.core.windows.net/vhds/vhdname.vhd
+                if ($osDiskUri) {
+                    $osDiskContainerName = $osDiskUri.Split('/')[-2]
+                    $osDiskStorageAcct = Get-AzStorageAccount | where { $_.StorageAccountName -eq $osDiskUri.Split('/')[2].Split('.')[0] }
+                    Write-Host "Remove unmanaged OS disk $osDiskUri"
+                    $osDiskStorageAcct | Remove-AzStorageBlob -Container $osDiskContainerName -Blob $osDiskUri.Split('/')[-1]
+                } else {
+                    Write-Host "Remove managed OS disk $($vm.StorageProfile.OsDisk.Name)"
+                    Get-AzDisk -DiskName $vm.StorageProfile.OsDisk.Name  | Remove-AzDisk -Force
+                }
+
+                foreach ($dataDisk in $vm.StorageProfile.DataDisks) {
+                    if ($dataDisk.Vhd.Uri) {
+                        $dataDiskStorageAcct = Get-AzStorageAccount -Name $uri.Split('/')[2].Split('.')[0]
+                        Write-Host "Remove unmanaged data disk $uri"
+                        $dataDiskStorageAcct | Remove-AzStorageBlob -Container $uri.Split('/')[-2] -Blob $uri.Split('/')[-1]
+                    }
+                    if ($dataDisk.ManagedDisk) {
+                        Write-Host "Remove managed data disk $($dataDisk.Name)"
+                        Get-AzDisk -DiskName $dataDisk.Name  | Remove-AzDisk -Force
```

Comment:
Is it 100% sure about disk removal?

Reply:
Frankly speaking - no, when hit azure backend service issue with small chance.
But we have cleanup pipeline Shital is working on, will clean up unattached disk and storage account regularly.